src/server/scripts/Northrend/zone_grizzly_hills.cpp

```diff
@@ -934,6 +935,254 @@ class spell_infected_worgen_bite : public SpellScriptLoader
+    SPELL_WARHEAD_FUSE = 49181
+};
+// 49107 - Vehicle: Warhead Fuse
+class spell_vehicle_warhead_fuse : public SpellScriptLoader
+{
+public:
+    spell_vehicle_warhead_fuse() : SpellScriptLoader("spell_vehicle_warhead_fuse") { }
+
+    class spell_vehicle_warhead_fuse_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_vehicle_warhead_fuse_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARHEAD_Z_CHECK) || !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_SEEKING_LUMBERSHIP) || !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_FUSE))
+                return false;
+            return true;
+        }
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            Unit* caster = GetCaster();
+
+            caster->CastSpell(caster, SPELL_WARHEAD_Z_CHECK, true);
+            caster->CastSpell(caster, SPELL_WARHEAD_SEEKING_LUMBERSHIP, true);
+            caster->CastSpell(caster, SPELL_WARHEAD_FUSE, true);
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_vehicle_warhead_fuse_SpellScript::HandleDummy, EFFECT_1, SPELL_EFFECT_SCRIPT_EFFECT);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_vehicle_warhead_fuse_SpellScript();
+    }
+};
+
+enum WarheadDenonate
+{
+    SPELL_PARACHUTE = 66154,
+    SPELL_TORPEDO_EXPLOSION = 49290,
+    NPC_ALLIANCE_LUMBERBOAT_EXPLOSIONS = 27689
+};
+// 49250 - Detonate
+class spell_warhead_detonate : public SpellScriptLoader
+{
+public:
+    spell_warhead_detonate() : SpellScriptLoader("spell_warhead_detonate") { }
+
+    class spell_warhead_detonate_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warhead_detonate_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_PARACHUTE) || !sSpellMgr->GetSpellInfo(SPELL_TORPEDO_EXPLOSION))
+                return false;
+            return true;
+        }
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            Unit* caster = GetCaster();
+            Player* player = GetHitPlayer();
+            if (!player)
+                return;
+
+            player->ExitVehicle();
+            float horizontalSpeed = 3.0f;
+            float verticalSpeed = 40.0f;
+            player->KnockbackFrom(caster->GetPositionX(), caster->GetPositionY(), horizontalSpeed, verticalSpeed);
+            caster->CastSpell(player, SPELL_PARACHUTE, true);
+
+            std::list<Creature*> explosionBunnys;
+            caster->GetCreatureListWithEntryInGrid(explosionBunnys, NPC_ALLIANCE_LUMBERBOAT_EXPLOSIONS, 90.0f);
+            for (std::list<Creature*>::const_iterator itr = explosionBunnys.begin(); itr != explosionBunnys.end(); ++itr)
+                (*itr)->CastSpell((*itr), SPELL_TORPEDO_EXPLOSION, true);
```

Comment:
missing brackets please, also you can use `auto` instead of `std::list<Creature*>::const_iterator`